packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -587,3 +593,39 @@ export class TelemetryNullLogger implements ITelemetryLogger {
     public sendErrorEvent(event: ITelemetryErrorEvent, error?: any): void {}
     public sendPerformanceEvent(event: ITelemetryPerformanceEvent, error?: any): void {}
 }
+
+/**
+ * Take in a event object, stringify any fields that are non-primitives, and return the new event object.
+ * @param event - Event with fields you want to stringify.
+ */
+function convertToBaseEvent(event: TelemetryEventTypes): void {
+    for (const key of Object.keys(event)) {
+        const filteredEventVal = convertToBasePropertyType(event[key]);
+        if (filteredEventVal !== null) {
+            event[key] = filteredEventVal;
+        }
+    }
+}
+
+/**
+ * Takes in parameter, if parameter is of primitive type, return the original value.
+ * If parameter is an array, stringify then return the result.
+ * @param x - parameter passed to validate/filter
+ */
+function convertToBasePropertyType(x: any): TelemetryEventPropertyType | null {
+    switch (typeof x) {
+        case "string":
+        case "number":
+        case "boolean":
+        case "undefined":
+            return x;
+        default:
+            if (!Array.isArray(x)) {
+                return null;
+            }
+            if (x.every((val) => typeof val === "boolean" || typeof val === "string" || typeof val === "number")) {
+                return JSON.stringify(x);
+            }
+            return null;
```

Comment:
can we return undefined here, instead of null? undefined already has semantics you want here - it's not going to get to telemetry (irrespective if convertToBaseEvent is going to filter it out or not)